src/roadrunner-symfony-nyholm/src/Runner.php

```diff
@@ -38,14 +51,69 @@ public function run(): int
         while ($request = $worker->waitRequest()) {
             try {
                 $sfRequest = $this->httpFoundationFactory->createRequest($request);
+
+                $sessionName = $this->sessionOptions['name'] ?? \session_name();
+                $requestSessionId = $sfRequest->cookies->get($sessionName, '');
+                \session_id($requestSessionId);
+
+                /** @var Response $sfResponse */
                 $sfResponse = $this->kernel->handle($sfRequest);
+
+                $hasRequestSession = $sfRequest->hasSession();
+                if ($hasRequestSession) {
+                    $session = $sfRequest->getSession();
+
+                    $sessionId = \session_id();
```

Comment:
Why are we using PHP's built-in sessions? In particular, these won't work across multiple servers.

Reply:
Symfony uses this also when used Redis or other Session handlers all are calling that method to get the session_id thats why I call it here. I was afraid accessing again the session a new session start is trigger so I did directly acces sthe `session_id` here:

https://github.com/symfony/symfony/blob/bbb0a694fbbd5a527b2236636271a47122b5d47c/src/Symfony/Component/HttpFoundation/Session/Storage/Proxy/AbstractProxy.php#L78

Reply:
But there are still open problems, so this is still WIP. Have a look at https://github.com/php-runtime/runtime/issues/46 for which things are currently not resolved.

Reply:
As described in the issue alternative solution would be which I personally would prefer that maybe symfony add an option to write the `cookie` at: https://github.com/symfony/symfony/blob/bbb0a694fbbd5a527b2236636271a47122b5d47c/src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php#L110 to `$request` object this would make less problems for us.